### ISSUE TITLE
Issues with Updating Tasks and Dismissing events

### ISSUE THREAD

**Issue by greycel** [reflexsoar/reflex-api#455]
Hi [n3tsurge](https://github.com/n3tsurge), 

Using Reflex Version `2023.09.28-rc0`, and noticed below issues:

1. As an admin or case-owner, unable to assign a new task to a user - changes are reverting back after reloading the page and the designated task state changes back to `unassigned`.
2. The case task gets assigned only after clicking the start task button, regardless of whom the case owner assigns the task to from dropdown, it always gets assigned to the logged user.
3. If `analyst1` has started the task, as the case owner `admin` couldn't reassign it to another user `analyst2` - new changes are reverting back to earlier after reloading the page.
4. Dismissing events directly from the event queue page isn't working, 
5. The Observables actions menu item `copy value` isn't working

There are no errors in the backend logs, all the requests were successful.

**Comment by n3tsurge** [comment]
Linking this commit from UI https://github.com/reflexsoar/reflex-ui/commit/e394313b0774ed56693cef2e88a30e32af96f64f

**Comment by n3tsurge** [comment]
I am unable to replicate item 4 on this list.  Items 1,2,3 and 5 will be fixed in a release tonight

**Comment by greycel** [comment]
Thank you for the update...

Reg item 4 - From the event queue page, tried dismissing the event with a comment directly by clicking on the dismiss icon. 
HTTP PUT request was successful but the event status remains new and unchanged.

**Comment by n3tsurge** [comment]
Unable to replicate item 4.  I would confirm that your `reflex-events` index is not in read-only state.

**Comment by greycel** [comment]
Reg item 4 - Yeah, the index `reflex-events` is writeable.

Reg item-5 `copy value` - Context menu fix is working fine but the `click to copy` option over the observable isn't working as expected.

![image](https://github.com/reflexsoar/reflex-api/assets/11851504/d1076351-ac26-4198-86e4-5882dafa46da)

Steps to Reproduce:
- Click on an observable like `user` and then use the `click to copy` option that appears over the observable value to copy its value.
- Choose a different observable this time like `hostname`, and notice that the observable value remains unchanged - still shows the `user` as its value even though the selected observable is `hostname`, appears like the variable value isn't updating.

**Comment by n3tsurge** [comment]
Item 5 should be fixed in the next patch, it's mostly a visual error at the moment.

Do you have any additional information for item 4, I am unable to replicate this in any environment we have running the latest release v2023.10.25-rc0

**Comment by greycel** [comment]
Unfortunately, no errors were found for ~~item 5~~ item 4 in either the UI or the backend logs.
Currently using version `2023.09.28-rc0`, will test with the newer release and let you know. Thank you...

**Comment by n3tsurge** [comment]
Do you mean item 4?  Like when you go to dismiss an event just nothing happens? The page reloads the event isn't dismissed?

**Comment by greycel** [comment]
Yeah, I mean Item 4 - Dismissing events directly from the event queue page isn't working,
